### DISCUSSION TITLE
Using webhooks and API to write custom scripts

### DISCUSSION THREAD

**Discussion by arijoury** [chatwoot/chatwoot#5392] [Q&A]
Hi, I would like to know if it’s possible to add custom scripts to a Heroku deployment of Chatwoot. 

I'm working with a self-hosted version of Chatwoot which we deployed to Heroku a few months ago. It integrates with our Telegram Bot. We’re really happy with the product but we’d like to automate a substantial part of the conversations now. We don't need any AI/NLP solutions; I have written a Python script that takes care of getting the right information from our users. It's appended here as a txt file since GitHub doesn't like .py extensions. (The name of the script is terrible because it was thought for a prototype at first).

The script works in a regular Telegram bot. However, I don’t know how to add this script to our existing deployment with Heroku. I am able to deploy a different Telegram bot with the script to Heroku without the Chatwoot integration, but I don’t know how to add the script to our original bot which uses Chatwoot.

I realize that this might not be a trivial question to answer. Nevertheless I hope you can help us out there!
[lgbtukraineshitbot.txt](https://github.com/chatwoot/chatwoot/files/9482575/lgbtukraineshitbot.txt)

**Comment by vishnu-narayanan** [comment]
@arijoury Are you looking to use the logic in this script to automate all the conversations you receive in your self-hosted instance? Which channels/inboxes are your using?

**Reply by arijoury** [reply]
@vishnu-narayanan Yes, the idea is that the script asks all the basic questions that are needed from the user, and if needed the conversation gets handed over to a human operator after the script is done. 
We're just using one inbox which is called lgbtukrainebot. Our only channel is a Telegram bot, https://t.me/lgbtukrainebot.
Let me know if you need any other information!

**Comment by sojan-official** [comment]
You will have to configure the script as a cloud function or a separate deployment.   Then use the Webhook integration to subscribe to the conversation events in the inbox. Based on the events, you can respond to Chatwoot server using the APIs

ref: https://github.com/chatwoot/google-cloud-functions-demo

**Reply by arijoury** [reply]
Thanks for the tip, Sojan! I'll check this out and if it works I'll mark your response as an answer.

**Reply by arijoury** [reply]
So I've had a look at it, but I'm not sure if I can make the webhook integration point at this script? What I would need would be a webhook that executes my Python script when a new conversation is created, then hands the conversation over to the human agents when the script has finished running (i.e. the script arrived at its last question). Do you think this is possible?
